### PR TITLE
TNG-747 identify and handle not-found

### PR DESCRIPTION
**Pull request by christian-breidler**

### CODE REVIEW

File: src/store/index.ts [hunk 1]
```diff
@@ -144,10 +150,14 @@ export function getFSXAModule<R extends RootState>(
...
           );
-          let navigationData = await fsxaAPI.fetchNavigation(
-            path || null,
-            payload.defaultLocale,
-          );
+
+          let navigationData = await fsxaAPI
+            .fetchNavigation(path || null, payload.defaultLocale)
+            .catch(reason => {
+              if (isNotFoundError(reason)) return null;
```
Comment (henczi-espirit) [inline src/store/index.ts:157]: Can be replaced by `if (reason?.message === FSXAApiErrors.NOT_FOUND)`

Comment (henczi-espirit) [reply src/store/index.ts:157]: Why do we want to make the navigationData null in the first place? 
In line 162 we will refetch the navigation.

**Review by henczi-espirit — CHANGES_REQUESTED** [review]
Please answer the comments!

File: src/store/index.ts (hunk 1 above)
Comment (christian-breidler) [reply src/store/index.ts:157]: > Can be replaced by `if (reason?.message === FSXAApiErrors.NOT_FOUND)`

It is possible that `reason` is e.g. a String and this results in a Syntax Error for the optional chaining operator.


> Why do we want to make the navigationData null in the first place? In line 162 we will refetch the navigation.

We need a falsy return value here to trigger line 162, `null` is most readable.

**Pull request merged**